tmt/steps/provision/bootc.py

```diff
@@ -386,24 +388,27 @@ def go(self, *, logger: Optional[tmt.log.Logger] = None) -> None:
         data = BootcData.from_plugin(self)
         data.show(verbose=self.verbosity_level, logger=self._logger)
 
-        if self._rootless:
+        if self._rootless and not self.is_dry_run:
             self._init_podman_machine()
 
         # Use provided container image
+        containerimage = ''
```

Comment:
An empty string should be a sign of "unknown", "unset" or "not initialized", `None` is more fitting and type checkers can trace it. I think there are two options:

* `containerimage: Optional[str] = None`, or
* `containerimage: str` - no value assigned, just declaring the name and type.

Then continue as you did. I think the latter option might be better, is clearly expected by `GuestBootc`, and type checkers and linters will tell you if it's left uninitialized somehow. Setting it to an empty string may be masking the error of passing the empty string where a string is expected, which would hardly be a valid value.